src/main/java/rx/swing/sources/ChangeEventSource.java

```diff
@@ -0,0 +1,114 @@
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public enum ChangeEventSource { ; // no instances
+
+    /**
+     * Creates an observable corresponding to change events (e.g. progressbar value changes).
+     *
+     * Due to the lack of a common interface in Java (up to at least version 8), the implementation
+     * is generic and uses internally reflection to add and remove it's {@link ChangeListener}'s.
+     * The contract is therefor that the given parameter object MUST have the typical two public methods "addChangeListener"
+     * (like {@link javax.swing.JProgressBar#addChangeListener(ChangeListener)}) and "removeChangeListener"
+     * (like {@link javax.swing.JProgressBar#removeChangeListener(ChangeListener)}).
+     *
+     * For more info to change listeners and events see <a href="https://docs.oracle.com/javase/tutorial/uiswing/events/changelistener.html">
+     * How to Write a Change Listener</a>.
+     *
+     * @param changeEventSource
+     * 		The object to register the observable for.
+     * @return Observable emitting the change events.
+     * @throws IllegalArgumentException if the given parameter object has not the needed signature
+     */
+    public static Observable<ChangeEvent> fromChangeEventsOf(final Object changeEventSource) {
+        checkHasChangeListenerSupport(changeEventSource);
+        return Observable.create(new OnSubscribe<ChangeEvent>() {
+            @Override
+            public void call(final Subscriber<? super ChangeEvent> subscriber) {
+                final ChangeListener listener = new ChangeListener() {
+
+                    @Override
+                    public void stateChanged(final ChangeEvent event) {
+                        subscriber.onNext(event);
+                    }
+
+                };
+                addChangeListener(changeEventSource, listener);
+                subscriber.add(Subscriptions.create(new Action0() {
+                    @Override
+                    public void call() {
+                        removeChangeListener(changeEventSource, listener);
+                    }
+                }));
+            }
+        }).subscribeOn(SwingScheduler.getInstance())
+                .unsubscribeOn(SwingScheduler.getInstance());
+    }
+
+    private static void checkHasChangeListenerSupport(Object object) {
+        checkPublicMethodExists(object, "addChangeListener", ChangeListener.class);
+        checkPublicMethodExists(object, "removeChangeListener", ChangeListener.class);
+    }
+
+    private static void checkPublicMethodExists(Object object, String methodName, Class<?>... parameterTypes) {
+        try {
+            Method method = object.getClass().getMethod(methodName, parameterTypes);
+            if (!Modifier.isPublic(method.getModifiers())) {
+                throw new IllegalArgumentException(
+                        "'" + object + "' has not the expected signature to support change listeners in "
```

Comment:
This message relies on the object having a sane implementation of toString(), perhaps, it would be more informative to rather to use `object.getClass().getSimpleName()`. This comment applies to the other instances within this file of using object in a message.